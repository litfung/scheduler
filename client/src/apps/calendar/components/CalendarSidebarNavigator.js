import React, { useState, useEffect } from 'react';
import { FaAngleLeft, FaAngleRight } from 'react-icons/fa';

import { month_name_dictionary, day_name_dictionary } from '../../../utils/constants';

const CalendarSidebarNavigator = (props) => {
	const { month, year, handleNavigation } = props;

	return (
		<div className="CalendarSidebarNavigator-c-00 Flex-Class-Row-Space-Between">
			<p>{month_name_dictionary[month]}</p>
			<div className="Flex-Class-Row-Center">
				<button onClick={() => handleNavigation(false)}>
					<FaAngleLeft style={{ margin: 'auto 2px 0px auto' }} />
				</button>
				<button onClick={() => handleNavigation(true)}>
					<FaAngleRight style={{ margin: 'auto auto 0px 2px' }} />
				</button>
			</div>
		</div>
	);
};

export default CalendarSidebarNavigator;
