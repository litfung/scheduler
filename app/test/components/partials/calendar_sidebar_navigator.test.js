import React from 'react';
import { shallow } from 'enzyme';

import CalendarSidebarNavigator from '../../../src/client/components/partials/calendar/calendar_sidebar_navigator';

let wrapper, handleNavigation, navigateToNow;

beforeEach(() => {
  handleNavigation = jest.fn();
  navigateToNow = jest.fn();

  wrapper = shallow(
    <CalendarSidebarNavigator
      miniCalendarState={{ year: 2019, month: 0 }}
      handleNavigation={handleNavigation}
      navigateToNow={navigateToNow}
    />
  );
});

test('Snapshot for CalendarSidebarNavigator Component', () => {
  expect(wrapper).toMatchSnapshot();
});
