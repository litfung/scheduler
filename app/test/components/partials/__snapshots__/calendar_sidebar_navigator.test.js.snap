// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for CalendarSidebarNavigator Component 1`] = `
<div
  className="calendar-sidebar-navigation-000"
>
  <p>
    January
    Â 
    2019
  </p>
  <div>
    <button
      className="sidebar-navigation-001-nav-btn"
      name="Navigate to Previous Month"
      onClick={[Function]}
      style={
        Object {
          "marginRight": "2px",
        }
      }
    >
      &lt;
    </button>
    <button
      className="sidebar-navigation-001-nav-btn"
      name="Navigate to Next Month"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "2px",
        }
      }
    >
      &gt;
    </button>
  </div>
</div>
`;
